### PR TITLE
Nucleo_l4r5zi: Enable LPUART_1 as console

### LINKED ISSUE #12092: Default Virtual COM port on nucleo_l4r5zi is lpuart1
**Describe the bug**
According to board user manual, VPC for nucleo_l4r5zi is  lpuart1.
Cf http://www.st.com/resource/en/user_manual/dm00368330.pdf

**To Reproduce**
1. mkdir build; cd build
2. cmake -DBOARD=board\_xyz
3. make

**Expected behavior**
Change zephyr,console and zephyr,shell-uart to lpuart1

**Impact**
No console output

**Screenshots or console output**
NA

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used

**Additional context**
NA

### PR DESCRIPTION
**Pull request by erwango**
LPUART_1 is default serial port used on ST-Link VPC for board nucleo_l4r5zi.
It is configured to use Port G pins which require dedicated power supply.

This PR does the needful for port G power activation, LPUART_1 configuration on board and some clean up in STM32 UART Kconfig symbols.

Fixes #12092

### CODE REVIEW

**Comment by erwango** [comment]
@psidhu, please review, as you were the one pushing this board
@gon1332, you might reuse this for nucleo_l496zg
@cybertale, please take it into account in #12024

**Comment by cybertale** [comment]
@erwango OK, I'll see what I can do.

File: drivers/gpio/gpio_stm32.c [hunk 1]
```diff
@@ -197,6 +197,20 @@ static int gpio_stm32_init(struct device *device)
 		return -EIO;
 	}
 
+#ifdef PWR_CR2_IOSV
+	if (cfg->port == STM32_PORTG) {
+		/* Port G[15:2] tequires external power supply */
```
Comment (gon1332) [inline drivers/gpio/gpio_stm32.c]: typo on requires

File: boards/arm/nucleo_l4r5zi/nucleo_l4r5zi.dts [hunk 2]
```diff
@@ -70,7 +70,7 @@ arduino_serial: &usart3 {};
 	status = "ok";
 };
 
-&usart3 {
+&lpuart1 {
 	current-speed = <115200>;
```
Comment (psidhu) [inline boards/arm/nucleo_l4r5zi/nucleo_l4r5zi.dts]: You should add another node, note replace usart3. usart3 is used I think for the Arduino shield.

**Review by psidhu — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_l4r5zi/nucleo_l4r5zi.dts (hunk 2 above)
Comment (erwango) [reply boards/arm/nucleo_l4r5zi/nucleo_l4r5zi.dts]: Right, this is actually a hack that went in. Fixed

File: drivers/gpio/gpio_stm32.c (hunk 1 above)
Comment (erwango) [reply drivers/gpio/gpio_stm32.c]: ack

**Comment by gon1332** [comment]
With a first check on basic/thread on my NUCLEO L496ZG board, I cannot see any output on console.

I've checked both `/dev/ttyACMx` and `/dev/ttyUSBx` just to be sure.

**Comment by gon1332** [comment]
> With a first check on basic/thread on my NUCLEO L496ZG board, I cannot see any output on console.
> 
> I've checked both `/dev/ttyACMx` and `/dev/ttyUSBx` just to be sure.

Oh wait! You haven't used USB ACM. I will recheck.

**Comment by gon1332** [comment]
> > With a first check on basic/thread on my NUCLEO L496ZG board, I cannot see any output on console.
> > I've checked both `/dev/ttyACMx` and `/dev/ttyUSBx` just to be sure.
> 
> Oh wait! You haven't used USB ACM. I will recheck.

I didn't work. The output is empty.
The strange thing is that when I choose LPUART_1 from DTS for console output, the application with which I flash the board is not running.

**Comment by erwango** [comment]
> The strange thing is that when I choose LPUART_1 from DTS for console output, the application with which I flash the board is not running.

I've tested these changes on top of you branch and worked. Did you update Kconfig.defconfig?

**Comment by gon1332** [comment]
@erwango I've checked your fix, with success at this time. How do you see it merged in my PR (https://github.com/zephyrproject-rtos/zephyr/pull/11265)?

Which one will get merged first? Do you want me to have the necessary changes ready?

**Comment by erwango** [comment]
@gon1332 , let try to merge this one first. Can you approve it if finally validated?
@psidhu, txs for the review, I fixed your point, can you review again?

**Review by gon1332 — APPROVED** [review]

**Review by psidhu — APPROVED** [review]
LGTM

**Comment by gon1332** [comment]
@erwango Nice! 2 out of 2.
Is it time yet to merge this PR?

**Comment by erwango** [comment]
@galak, this one is ready for merge if you have some time

**Review by galak — APPROVED** [review]

**Pull request merged**